File: pyround1/350.两个数组的交集-ii.py
#
# @lc app=leetcode.cn id=350 lang=python
#
# [350] 两个数组的交集 II
#

# @lc code=start
class Solution(object):
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        dic = defaultdict(int)
        r = []
        for i in nums1: 
            dic[i] += 1
        for i in nums2:
            if dic[i] != 0:
                r.append(i)
                dic[i] -= 1
        return r
        
# @lc code=end

File: 350.两个数组的交集-ii.cpp
/*
 * @lc app=leetcode.cn id=350 lang=cpp
 *
 * [350] 两个数组的交集 II
 */

// @lc code=start
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int, int>com;
        vector<int> res;
        if(nums1.size() > nums2.size()){
            swap(nums1, nums2);
        }
        for(auto num : nums1){
            com[num] += 1;
        }
        for(auto num : nums2){
            if(com[num]){
                res.push_back(num);
                com[num] -=1;
            }
        }
        return res;
    }
};
// @lc code=end

